batch/batch/aioclient.py

```diff
@@ -0,0 +1,328 @@
+        }])
+
+        container = {
+            'image': image,
+            'name': 'main'
+        }
+        if command:
+            container['command'] = command
+        if args:
+            container['args'] = args
+        if env:
+            container['env'] = env
+        if ports:
+            container['ports'] = [{
+                'containerPort': p,
+                'protocol': 'TCP'
+            } for p in ports]
+        if resources:
+            container['resources'] = resources
+        if volumes:
+            container['volumeMounts'] = [v['volume_mount'] for v in volumes]
+        spec = {
+            'containers': [container],
+            'restartPolicy': 'Never'
+        }
+        if volumes:
+            spec['volumes'] = [v['volume'] for v in volumes]
+        if tolerations:
+            spec['tolerations'] = tolerations
+        if security_context:
+            spec['securityContext'] = security_context
+        if service_account_name:
+            spec['serviceAccountName'] = service_account_name
+
+        doc = {
+            'spec': spec,
+            'parent_ids': parent_ids,
+            'always_run': always_run
+        }
+        if attributes:
+            doc['attributes'] = attributes
+        if batch_id:
+            doc['batch_id'] = batch_id
+        if callback:
+            doc['callback'] = callback
+        if scratch_folder:
+            doc['scratch_folder'] = scratch_folder
+        if input_files:
+            doc['input_files'] = input_files
+        if output_files:
+            doc['output_files'] = output_files
+        if copy_service_account_name:
+            doc['copy_service_account_name'] = copy_service_account_name
+
+        response = await self._session.post(self.url + '/jobs/create', json=doc)
+        j = await response.json()
+
+        return Job(self,
+                   j['id'],
+                   attributes=j.get('attributes'),
+                   parent_ids=j.get('parent_ids', []))
+
+    async def _get_job(self, id):
+        response = await self._session.get(self.url + '/jobs/{}'.format(id))
+        return await response.json()
+
+    async def _get_job_log(self, id):
+        response = await self._session.get(self.url + '/jobs/{}/log'.format(id))
+        return await response.json()
+
+    async def _delete_job(self, id):
+        response = await self._session.delete(self.url + '/jobs/{}/delete'.format(id))
+        return await response.json()
+
+    async def _cancel_job(self, id):
+        response = await self._session.post(self.url + '/jobs/{}/cancel'.format(id))
+        return await response.json()
+
+    async def _get_batch(self, batch_id):
+        response = await self._session.get(self.url + '/batches/{}'.format(batch_id))
+        return await response.json()
+
+    async def _close_batch(self, batch_id):
+        response = await self._session.post(self.url + '/batches/{}/close'.format(batch_id))
+        return await response.json()
```

Comment:
There isn't a delete batch. Also, the `api.py` route for `delete_batch` is broken.

Reply:
Good catch, fixed.